### PR TITLE
[feature](new-scan) support transactional insert in new scan framework

### PR DESCRIPTION
**Pull request by morningman**
# Proposed changes

Issue Number: close #xxx

## Problem summary

1.
Support running transactional insert operation with new scan framework. eg:
```
admin set frontend config("enable_new_load_scan_node" = "true");
begin;
insert into tbl1 values(1,2);
insert into tbl1 values(3,4);
insert into tbl1 values(5,6);
commit;
```

2. Add some limitation to transactional insert
    1. Do not support non-literal value in insert stmt

3. Fix some issue about array type:
    1. Forbid cast other non-array type to NESTED array type, it may cause BE crash.
    2. Add `getStringValueForArray()` method for Expr, to get valid string-formatted array type value. 

4. Add `useLocalSessionState=true` in regression-test jdbc url
    without this config, the jdbc driver will send some init cmd each time it connect to server, such as
    `select @@session.tx_read_only`.
    But when we use transactional insert, after `begin` command, Doris do not support any other type of
    stmt except for `insert`, `commit` or `rollback`.
    So adding this config to let the jdbc NOT send cmd when connecting.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
3. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
6. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
7. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
8. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/analysis/NullLiteral.java [hunk 1]
```diff
@@ -101,6 +101,11 @@ public String getStringValue() {
         return "NULL";
     }
 
+    @Override
+    public String getStringValueForArray() {
```
Comment (pengxiangyu) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/NullLiteral.java:105]: null?  not return ""？

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 38.08 seconds
 load time: 573 seconds
 storage size: 17154821321 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221101091724_clickbench_pr_37260.html

File: fe/fe-core/src/main/java/org/apache/doris/analysis/NullLiteral.java [hunk 2]
```diff
@@ -101,6 +101,11 @@ public String getStringValue() {
...
+        return "null";
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/NullLiteral.java:106]: the null value inside an array is represented as "null", for exampe:
`[null, null]`.
Not same as other primitive type to represent as `\N`.
I will add some comment to this method.

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**